File: sdks/dotnet/src/Croniq.Runner.Sdk/Protocol/AckRequest.cs
using System.Text.Json.Serialization;

namespace Croniq.Runner.Sdk.Protocol;

internal sealed record AckRequest(
    [property: JsonPropertyName("runner_id")] string RunnerId,
    [property: JsonPropertyName("execution_id")] string ExecutionId,
    [property: JsonPropertyName("status")] string Status,
    [property: JsonPropertyName("error"), JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)] string? Error,
    [property: JsonPropertyName("duration_ms"), JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)] long? DurationMs,
    [property: JsonPropertyName("attempt")] int Attempt);

File: sdks/java/core/src/main/java/io/croniq/runner/protocol/AckRequest.java
package io.croniq.runner.protocol;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * POST /v1/work/ack request body. {@code status} is the string {@code "success"}
 * or {@code "failure"} — see {@link Status} for the canonical constants.
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
public record AckRequest(
        @JsonProperty("runner_id") String runnerId,
        @JsonProperty("execution_id") String executionId,
        @JsonProperty("status") String status,
        @JsonProperty("error") String error,
        @JsonProperty("duration_ms") Long durationMs,
        @JsonProperty("attempt") int attempt) {

    public static final class Status {
        public static final String SUCCESS = "success";
        public static final String FAILURE = "failure";

        private Status() {}
    }
}
